runtime/util/romclasswalk.c

```diff
@@ -46,6 +46,9 @@ static void allSlotsInBytecodesDo (J9ROMClass* romClass, J9ROMMethod* method, J9
 static void allSlotsInCPShapeDescriptionDo (J9ROMClass* romClass, J9ROMClassWalkCallbacks* callbacks, void* userData);
 static void allSlotsInCallSiteDataDo (J9ROMClass* romClass, J9ROMClassWalkCallbacks* callbacks, void* userData);
 static UDATA allSlotsInMethodParametersDataDo(J9ROMClass* romClass, U_8* cursor, J9ROMClassWalkCallbacks* callbacks, void* userData);
+#if defined(J9VM_OPT_METHOD_HANDLE) && !defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+static void allSlotsInVarHandleMethodTypeLookupTableDo(J9ROMClass* romClass, J9ROMClassWalkCallbacks* callbacks, void* userData);
+#endif /* defined(J9VM_OPT_METHOD_HANDLE) && !defined(J9VM_OPT_OPENJDK_METHODHANDLE) */
```

Comment:
- `varHandleMethodTypeLookupTable` only exists in the OpenJ9 MH impl, which is enabled via `J9VM_OPT_METHOD_HANDLE`.
- `J9VM_OPT_OPENJDK_METHODHANDLE` refers to the OJDK MH impl and it doesn't use `varHandleMethodTypeLookupTable`.
- Whenever `J9VM_OPT_METHOD_HANDLE` is enabled, `J9VM_OPT_OPENJDK_METHODHANDLE` is disabled. Both flags can never be enabled together; otherwise, the builds will break.
- We should only associate theaw changes with the corresponding implementation: `J9VM_OPT_METHOD_HANDLE` (OpenJ9 MH impl). 
```suggestion
#if defined(J9VM_OPT_METHOD_HANDLE)
static void allSlotsInVarHandleMethodTypeLookupTableDo(J9ROMClass* romClass, J9ROMClassWalkCallbacks* callbacks, void* userData);
#endif /* defined(J9VM_OPT_METHOD_HANDLE) */
```